### ISSUE TITLE
How does the code in the psconv.py file achieve the different dilation rates written in the paper?

### ISSUE THREAD

**Issue by xlh1135355585** [d-li14/PSConv#5]
How does the code in the psconv.py file achieve the differentdilation rates written in the paper?

**Comment by doitslow** [comment]
> How does the code in the psconv.py file achieve the differentdilation rates written in the paper?

I think the different dilation rates are done through the three different convs. However, I find it very difficult to understand how the three conv results add up together and result in the 1,2,1,4 cyclic dilation rates.  I can see how half of the weights are been set off in the conv with no dilation. But what is the reordering of channels in shift conv for? @d-li14 Possible to take some time to do a little bit elaboration? It would be very appreciated.

**Comment by urbaneman** [comment]
> > How does the code in the psconv.py file achieve the differentdilation rates written in the paper?
> 
> I think the different dilation rates are done through the three different convs. However, I find it very difficult to understand how the three conv results add up together and result in the 1,2,1,4 cyclic dilation rates. I can see how half of the weights are been set off in the conv with no dilation. But what is the reordering of channels in shift conv for? @d-li14 Possible to take some time to do a little bit elaboration? It would be very appreciated.

I have the same question. 
Look forward to the author's answer. @d-li14 

https://github.com/d-li14/PSConv/blob/9afde67f43025bd5ae59e1ff1cc5a04c0744a5bd/mmdet/models/utils/psconv.py#L21

**Comment by Margrate** [comment]
but the function didn't use the line: self.mask[i * _out_channels: (i + 1) * _out_channels, i * _in_channels: (i + 1) * _in_channels, : , :] = 
def forward(self, x):
        x1, x2 = x.chunk(2, dim=1)
        x_shift = self.gwconv_shift(torch.cat((x2, x1), dim=1))
        return self.gwconv(x) + self.conv(x) + x_shift

just add three different conv.  It‘s not the same with paper

**Comment by big-xiao** [comment]
Because this code implementation is fake!The author of this code has been found to have falsified papers！！！I wasted two nights studying this shit！

**Comment by xlh1135355585** [comment]
This should be compiled by oneself with reference to other people's thoughts.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "d-li14/PSConv"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年12月17日(星期五) 下午4:08
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [d-li14/PSConv] How does the code in the psconv.py file achieve the different dilation rates written in the paper? (#5)





 
Because this code implementation is fake!The author of this code has been found to have falsified papers！！！I wasted two nights studying this shit！
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by big-xiao** [comment]
You can search duo Li on Quora，You can know about the fraud of the author's paper

**Comment by XinzheGeng** [comment]
In my opinion,
x_shift = self.gwconv_shift(torch.cat((x2, x1), dim=1))  can achieve conv with dilation 4 (green part in paper section B fig. 4),
self.gwconv(x) can achieve conv with dilation 2 (red part in paper section B fig. 4),
self.conv(x) can achieve conv with dilation 1 (blue and yellow parts in paper section B fig. 4).
where the param 'dilation' for class PSConv2d have to be 2 instead of 1.
It's a may simple implementation for this method.
